nio/store/database.py

```diff
@@ -397,6 +397,13 @@ def save_encrypted_rooms(self, rooms):
                 LegacyEncryptedRooms.account
             ]).on_conflict_ignore().execute()
 
+    @use_database_atomic
```

Comment:
I don't think you need to use atomic here, at least I didn't find any peewee docs mentioning it. Do you have a rationale for using it?

Reply:
I'm not familiar with peewee, I saw the decorator on the method above to save rooms but I wasn't sure if I should use it. I removed it now.

Reply:
The save room method inserts many rows at once, that's when we need to use atomic, but you do need the use_database decorator. Sorry for not being clear on that.

Another thing, the get() method can raise an exception if there is no such room. While this normally shouldn't happen we should catch the error and do nothing with the db if someone tries to delete a non-existing room.

You could also use get_or_none() which returns None instead of raising the DoesNotExist exception.